src/Team.java

```diff
@@ -0,0 +1,57 @@
+import java.util.ArrayList;
+
+public class Team {
```

Comment:
Myslim si, ze Team ako take, by mohlo radsej byt trieda *immutable* - tj. objekt pocas svojej existencie nemoze menit stav.
V tom pripade konstruktor by vypadal takto: Team(name, matchCount, winCount, lossCount) a trieda by obsahovala len metdoy getX()
Na vycucnutie detailov z List<MatchStats> by som vytvoril novu triedu MatchAnalyzer, s metodou getStatsForTeam(teamName).
Ma to svoje vyhody, ked objekt ako Team nemoze byt v nekonzistentom stavu pocas svojej existencie - ako priklad by som mohol uvest pripad, ze pri zpracovani zoznamu zapasov vo iteracii x objekt Team je v nekonzistentnom stave - obsahuje atributy, ktore vyjadruju nejaky medzistav. Dobry navrh by nemal take stavy objektov dovolit.

Reply:
Tomuto se priznam moc nerozumim. Rekneme, ze ten seznam je tabulka za nejake obdobi a vsledky pro kazdy tym to ulozi do objektu daneho tymu.
Kvuli tomu jsem musel vytvorit metody "addX"

Kdybych nemel ukladat vysledky do objektu "Team", tak kam?

Reply:
Do premennej. Treba berme metodu z prikladu vys: new `TeamAnalyzer(matchList).analyzeTeam(team);`
Na zaciatku vytvoris premenne `int matchesWon, matchesLost, matchesDraw`, pocitas si ich, a na konci vytvoris objekt `new Team(matchesWon, matchesLost, matchesDraw)`, ktory uz stav menit nemoze.

Reply:
Uz vidim, kde je trosku rozdiel vo vnimani ukolu: ty vypocitas vysledky stejne pre kazdy tym, i kdyz chces vypisovat len jeden tym. Moj navrh pocita s tym, ze chceme vysledky len jednoho tymu. Toto je otazka zadania - ked chces udrzat statistiky pre kazdy tym (a treba ulozit to do suboru), tak by som moj navrh na metodu main() zmenil takto:
```
Scanner inputScanner = new Scanner(System.in);

String fileName = promptFileName(inputScanner);
List<MatchStats> matchList = new MatchAnalyzer().createMatchList(fileName);

String teamName = promptTeamName(inputScanner);
Map<String, Team> teamMap = new TeamAnalyzer(matchList).analyzeTeams();
printTeamStats(teamMap.get(teamName));

inputScanner.close();
```